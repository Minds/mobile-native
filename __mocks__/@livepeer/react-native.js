// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
