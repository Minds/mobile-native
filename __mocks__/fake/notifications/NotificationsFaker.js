function load() {
  let notifications = [
    {
      uuid: '4a427b80-c53d-11eb-a497-93e2755f275f',
      urn:
        'urn:notification:968187695744425997-4a427b80-c53d-11eb-a497-93e2755f275f',
      to_guid: '968187695744425997',
      from_guid: '100000000000000519',
      from: {
        guid: '100000000000000519',
        type: 'user',
        subtype: false,
        time_created: '1623077876',
        time_updated: false,
        container_guid: false,
        owner_guid: false,
        site_guid: false,
        access_id: false,
        tags: [],
        nsfw: [],
        nsfw_lock: [],
        allow_comments: false,
        name: 'Minds',
        username: 'minds',
        icontime: '1623077876',
        legacy_guid: false,
        featured_id: false,
        banned: 'no',
        ban_reason: false,
        website: false,
        briefdescription: '',
        gender: false,
        city: false,
        merchant: false,
        boostProPlus: false,
        fb: false,
        mature: 0,
        monetized: false,
        signup_method: false,
        social_profiles: [],
        feature_flags: false,
        programs: [],
        plus: false,
        hashtags: false,
        verified: false,
        founder: false,
        disabled_boost: false,
        boost_autorotate: true,
        categories: [],
        wire_rewards: null,
        pinned_posts: [],
        is_mature: false,
        mature_lock: false,
        last_accepted_tos: 1558597098,
        opted_in_hashtags: 0,
        last_avatar_upload: '0',
        canary: false,
        theme: false,
        toaster_notifications: true,
        mode: 0,
        btc_address: '',
        surge_token: '',
        hide_share_buttons: false,
        allow_unsubscribed_contact: false,
        dismissed_widgets: [],
        liquidity_spot_opt_out: 0,
        urn: 'urn:user:100000000000000519',
        subscribed: true,
        subscriber: false,
        boost_rating: 1,
        pro: false,
        pro_published: false,
        rewards: false,
        p2p_media_enabled: false,
        is_admin: false,
        onchain_booster: 0,
        email_confirmed: true,
        eth_wallet: '',
        rating: 1,
        disable_autoplay_videos: false,
        yt_channels: [],
      },
      entity_urn: 'urn:user:968187695744425997',
      entity: {
        guid: '968187695744425997',
        type: 'user',
        subtype: false,
        time_created: '1556209934',
        time_updated: false,
        container_guid: '0',
        owner_guid: '0',
        site_guid: false,
        access_id: '2',
        tags: ['technology', 'crypto', 'blockchain', 'music', 'art'],
        nsfw: [],
        nsfw_lock: [],
        allow_comments: false,
        name: 'Juan',
        username: 'juanmsolaro',
        language: 'en',
        icontime: '1613843786',
        legacy_guid: false,
        featured_id: false,
        banned: 'no',
        ban_reason: false,
        website: '',
        briefdescription: 'Full Stack Developer at @minds',
        gender: '',
        city: 'Colonia Avellaneda',
        merchant: {
          service: 'stripe',
          id: 'acct_1Gj1tZDf4yhop99j',
        },
        boostProPlus: false,
        fb: false,
        mature: 1,
        monetized: '',
        signup_method: false,
        social_profiles: [
          {
            key: 'linkedin',
            value: 'linkedin.com/in/juan-manuel-solaro-730a48184',
          },
        ],
        feature_flags: false,
        programs: [],
        plus: false,
        hashtags: false,
        verified: true,
        founder: false,
        disabled_boost: false,
        boost_autorotate: true,
        categories: [],
        wire_rewards: {
          description: '',
          rewards: {
            points: [],
            money: [
              {
                amount: '1',
                description: 'VIP',
              },
            ],
            tokens: [
              {
                amount: '1',
                description: 'Common Citizen',
              },
              {
                amount: '2',
                description: 'Adventurer',
              },
              {
                amount: '3',
                description: 'Hero',
              },
            ],
          },
        },
        pinned_posts: ['1234531362564866048'],
        is_mature: false,
        mature_lock: false,
        last_accepted_tos: 1558597098,
        opted_in_hashtags: 5,
        last_avatar_upload: '1565703100',
        canary: true,
        theme: 'dark',
        toaster_notifications: true,
        mode: 0,
        btc_address: '1M27rFgyxqGdbmyjABu7J3pqa2YPhFs4mp',
        surge_token:
          'eyJzZXJ2aWNlIjoiZ29vZ2xlIiwidG9rZW4iOiJkbzZ2LWhfWGhxMDpBUEE5MWJGQ3N0T3c4QzdPR3M0ZjZDZ3VzWWhma0JScGstMF9rN0xValQyUmJhRGtmTmJpZXlSSG1oRmRJeTlRRTlEY0tsV0ZPZFNxR3hfNU9YQnI3VXFUV3paMDFNZ1ZSYm9fVXM3WjZrQVpiX2ZtdDU2akh2LVdUNXk3RzM4ZlBIQ1QwSDZEdXU5RyJ9',
        hide_share_buttons: true,
        allow_unsubscribed_contact: false,
        dismissed_widgets: [
          'language-sidebar-2020',
          'discovery-disclaimer-2020',
        ],
        liquidity_spot_opt_out: 0,
        chat: true,
        urn: 'urn:user:968187695744425997',
        subscribed: false,
        subscriber: false,
        boost_rating: 1,
        pro: false,
        pro_published: false,
        rewards: true,
        p2p_media_enabled: false,
        is_admin: false,
        onchain_booster: 0,
        email_confirmed: true,
        eth_wallet: '0xd3004e3cbA40079c31F80F0C465aacc34588bd79',
        rating: 1,
        disable_autoplay_videos: true,
        yt_channels: [],
      },
      read: true,
      created_timestamp: 1622815259,
      type: 'token_rewards_summary',
      data: {
        tokens_formatted: '0.332',
        usd_formatted: '0.36',
      },
      merged_from_guids: [],
      merged_from: [],
      merged_count: 0,
      merge_key:
        '52161ea64224a6ac08d653f9dd40b6e568d55362945138174507bf566e9e5e63',
    },
  ];

  return {
    notifications,
    'load-next': 'aaaaaa',
    status: 'sucess',
  };
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function notificationsServiceFaker() {
  return { load: load };
}
