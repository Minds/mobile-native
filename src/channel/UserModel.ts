import { observable, action, runInAction } from 'mobx';

import { MINDS_CDN_URI, GOOGLE_PLAY_STORE } from '../config/Config';
import api from '../common/services/api.service';
import BaseModel from '../common/BaseModel';
import ChannelService from './ChannelService';
import sessionService from '../common/services/session.service';
import apiService from '../common/services/api.service';
import logService from '../common/services/log.service';
import { SupportTiersType } from '../wire/WireTypes';

//@ts-nocheck
export const USER_MODE_OPEN = 0;
export const USER_MODE_MODERATED = 1;
export const USER_MODE_CLOSED = 2;

/**
 * User model
 */
export default class UserModel extends BaseModel {
  programs;
  merchant;
  /**
   * Eth wallet
   */
  eth_wallet: string = '';
  disable_autoplay_videos?: boolean;
  sums;
  btc_address?: string;
  icontime!: string;
  username!: string;
  briefdescription!: string;
  city!: string;
  name!: string;
  is_admin = false;
  plus: boolean = false;
  verified: boolean = false;
  founder: boolean = false;
  rewards: boolean = false;
  last_accepted_tos: number = 0;
  subscriptions_count: number = 0;
  carousels?: Array<any>;
  dob?: string;

  tags: Array<string> = [];
  groupsCount: number = 0;

  /**
   * @var {boolean}
   */
  @observable blocked;

  /**
   * @var {number}
   */
  @observable subscribers_count;

  /**
   * @var {number}
   */
  @observable impressions;

  /**
   * @var {boolean}
   */
  @observable subscribed;

  /**
   * @var {boolean}
   */
  @observable mature_visibility = false;

  /**
   * @var {boolean}
   */
  @observable pending_subscribe = false;

  /**
   * @var {numeric}
   */
  @observable mode = 0;

  /**
   * @var {boolean}
   */
  @observable email_confirmed = false;

  @observable wire_rewards;

  @observable pro: boolean = false;

  onchain_booster: number = 0;

  /**
   * Confirm email
   * @param {Object} params
   */
  confirmEmail = async (params) => {
    // call any api endpoint with the param
    try {
      await apiService.get('api/v2/entities/', { urn: this.urn, ...params });
      this.setEmailConfirmed(true);
      return true;
    } catch (error) {
      return false;
    }
  };

  /**
   * Get the user icon time
   */
  getOwnerIcontime(): string {
    if (sessionService.getUser().guid === this.guid) {
      return sessionService.getUser().icontime;
    } else {
      return this.icontime;
    }
  }

  @action
  toggleMatureVisibility() {
    if (GOOGLE_PLAY_STORE) return;
    this.mature_visibility = !this.mature_visibility;
  }

  @action
  toggleSubscription = async () => {
    const value = !this.subscribed;
    this.subscribed = value;

    try {
      const metadata = this.getClientMetadata();
      await ChannelService.toggleSubscription(this.guid, value, metadata);
      UserModel.events.emit('toggleSubscription', this);
    } catch (err) {
      runInAction(() => {
        this.subscribed = !value;
      });
      throw err;
    }
  };

  @action
  async toggleBlock(value: boolean | null = null) {
    value = value === null ? !this.blocked : value;

    try {
      await ChannelService.toggleBlock(this.guid, value);
      this.blocked = value;
    } catch (err) {
      this.blocked = !value;
      logService.exception('[ChannelStore] toggleBlock', err);
    }
  }

  @action
  setMode(value) {
    this.mode = value;
  }

  @action
  setEmailConfirmed(value) {
    this.email_confirmed = value;
  }

  @action
  setTier(tier: SupportTiersType, type: 'usd' | 'tokens') {
    const wire_rewards = this.wire_rewards;
    if (!wire_rewards.rewards) {
      wire_rewards.rewards = {
        money: [] as SupportTiersType[],
        tokens: [] as SupportTiersType[],
      };
    }
    if (type === 'tokens') {
      if (!wire_rewards.rewards.tokens) {
        wire_rewards.rewards.tokens = [] as SupportTiersType[];
      }
      wire_rewards.rewards.tokens.push(tier);
    } else {
      if (!wire_rewards.rewards.money) {
        wire_rewards.rewards.money = [] as SupportTiersType[];
      }
      wire_rewards.rewards.money.push(tier);
    }
    this.wire_rewards = wire_rewards;
  }

  @action
  togglePro() {
    this.pro = !this.pro;
  }

  /**
   * Is admin
   */
  isAdmin() {
    return this.is_admin;
  }

  /**
   * current user is owner of the channel
   */
  isOwner = () => {
    return sessionService.getUser().guid === this.guid;
  };

  /**
   * Get banner source
   */
  getBannerSource() {
    if (this.carousels) {
      return {
        uri: this.carousels[0].src,
      };
    }
    return {
      uri: `${MINDS_CDN_URI}fs/v1/banners/${this.guid}/fat/${this.icontime}`,
      headers: api.buildHeaders(),
    };
  }

  /**
   * Get avatar source
   * @param {string} size
   */
  getAvatarSource(size = 'medium') {
    return {
      uri: `${MINDS_CDN_URI}icon/${this.guid}/${size}/${this.icontime}`,
      headers: api.buildHeaders(),
    };
  }

  /**
   * Has banner
   * @returns {boolean}
   */
  hasBanner() {
    return !!this.carousels;
  }

  /**
   * Is closed
   * @returns {boolean}
   */
  isClosed() {
    return this.mode === USER_MODE_CLOSED;
  }

  /**
   * Is open
   * @returns {boolean}
   */
  isOpen() {
    return this.mode === USER_MODE_OPEN;
  }

  /**
   * Is moderated
   * @returns {boolean}
   */
  isModerated() {
    return this.mode === USER_MODE_MODERATED;
  }

  /**
   * Is subscribed
   * @returns {boolean}
   */
  isSubscribed() {
    return !!this.subscribed;
  }

  /**
   * Has avatar
   */
  hasAvatar(): boolean {
    return this.icontime !== this.time_created;
  }

  /**
   * Request subscribe
   */
  async subscribeRequest() {
    if (this.pending_subscribe || this.mode !== USER_MODE_CLOSED) {
      return;
    }
    try {
      this.pending_subscribe = true;
      await apiService.put(`api/v2/subscriptions/outgoing/${this.guid}`);
    } catch (err) {
      this.pending_subscribe = false;
      logService.exception(err);
    }
  }

  /**
   * Cancel subscribe request
   */
  async cancelSubscribeRequest() {
    if (!this.pending_subscribe || this.mode !== USER_MODE_CLOSED) {
      return;
    }
    try {
      this.pending_subscribe = false;
      await apiService.delete(`api/v2/subscriptions/outgoing/${this.guid}`);
    } catch (err) {
      this.pending_subscribe = true;
      logService.exception(err);
    }
  }
}
