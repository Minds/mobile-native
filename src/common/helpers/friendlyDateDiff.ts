export default function friendlyDateDiff(
  value: string | number,
  reference: string | number | null = null,
  displaySuffix: boolean = true,
): any {
  if (!value) {
    return value;
  }

  let referenceDate = new Date();

  if (reference) {
    referenceDate = new Date(<string>reference);
  }

  const dateValue = new Date(<string>value);

  const suffix = displaySuffix ? 'ago' : '';

  if (dateValue >= referenceDate) {
    return '0s ' + suffix;
  }

  let differenceMs = referenceDate.getTime() - dateValue.getTime();
  let seconds = Math.floor(differenceMs / 1000);
  let minutes = Math.floor(seconds / 60);
  let hours = Math.floor(minutes / 60);
  let days = Math.floor(hours / 24);
  let weeks = Math.floor(days / 7);
  let years = Math.floor(weeks / 52);

  if (years > 0) {
    return `${years}y ${suffix}`;
  }

  if (weeks > 0) {
    return `${weeks}w ${suffix}`;
  }

  if (days > 0) {
    return `${days}d ${suffix}`;
  }

  if (hours > 0) {
    return `${hours}h ${suffix}`;
  }

  if (minutes > 0) {
    return `${minutes}m ${suffix}`;
  }

  return `${seconds}s ${suffix}`;
}
