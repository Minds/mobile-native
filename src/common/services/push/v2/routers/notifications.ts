import navigation from '../../../../../navigation/NavigationService';
import {
  error,
  parseByEntityType,
  parseComment,
  parseReward,
} from '../parsers';

class NotificationsRouter {
  notifications_routes = {
    friends: (data: any) =>
      navigation.push('Channel', { guid: data.json.entity_guid }),
    group_invite: (data: any) =>
      navigation.push('GroupView', { guid: data.json.entity_guid }),
    like: parseByEntityType,
    downvote: parseByEntityType,
    group_activity: parseByEntityType,
    feature: parseByEntityType,
    tag: parseByEntityType,
    remind: (data: any) =>
      navigation.push('App', {
        screen: 'Activity',
        params: { guid: data.json.entity_guid },
      }),
    comment: parseComment,
    rewards_reminder: parseReward,
    rewards_summary: parseReward,
  };

  navigate(data: any) {
    data.json.type in this.notifications_routes
      ? this.notifications_routes[data.json.type](data)
      : error(data, 'Unknown notification');
  }
}

export default new NotificationsRouter();
