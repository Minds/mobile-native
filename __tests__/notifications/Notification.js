import 'react-native';
import React from 'react';

import { notificationsServiceFaker } from '../../__mocks__/fake/notifications/NotificationsFaker';
import NotificationModel from '../../src/notifications/v3/notification/NotificationModel';
import { shallow } from 'enzyme';
import Notification from '../../src/notifications/v3/notification/Notification';
import UserModel from '../../src/channel/UserModel';
import friendlyDateDiff from '../../src/common/helpers/friendlyDateDiff';

jest.mock('react-native-system-setting');
jest.mock('react-native-silent-switch');
jest.mock('../../src/common/helpers/friendlyDateDiff.ts');

jest.mock('../../src/common/services/session.service.ts');

jest.mock(
  '../../src/notifications/v3/notification/useNotificationRouter',
  () => () => ({
    navToEntity: jest.fn,
  }),
);

jest.mock('react-native-system-setting', () => {
  return {
    getVolume: jest.fn(() => Promise.resolve()),
  };
});

describe('Notifications component', () => {
  let notification;
  beforeEach(() => {
    friendlyDateDiff.mockReturnValue('1w');
    let activityResponse = notificationsServiceFaker().load();
    const notif = activityResponse.notifications[0];
    notif.from = UserModel.create(notif.from);
    if (notif.merged_from && notif.merged_from.length > 0) {
      notif.merged_from = UserModel.createMany(notif.merged_from);
    }
    const model = NotificationModel.create(notif);
    notification = shallow(<Notification notification={model} />);
  });
  it('renders correctly', async () => {
    expect(notification).toMatchSnapshot();
  });
});
