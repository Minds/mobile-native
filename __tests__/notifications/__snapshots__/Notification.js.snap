// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications component renders correctly 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    [
      [
        {
          "flex": 1,
          "flexDirection": "column",
          "paddingHorizontal": 20,
          "paddingVertical": 16,
        },
        {
          "borderBottomWidth": 0.5,
        },
        {
          "borderColor": "#53565A",
        },
      ],
      {
        "alignSelf": "center",
        "maxWidth": 750,
        "width": "100%",
      },
    ]
  }
>
  <AnimatedView
    style={
      {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <AnimatedView
      style={
        {
          "flexDirection": "column",
          "marginRight": 28,
        }
      }
    >
      <preventDoubleTap(TouchableOpacity)
        onPress={[Function]}
      >
        <Image
          source={
            {
              "headers": {
                "App-Version": "3.8.0",
                "Cache-Control": "no-cache, no-store, must-revalidate",
                "Content-Type": "application/json",
                "Pragma": "no-cache",
                "no-cache": "1",
              },
              "uri": "https://cdn.minds.com/icon/100000000000000519/medium/1623077876",
            }
          }
          style={
            {
              "borderRadius": 18.5,
              "height": 37,
              "width": 37,
            }
          }
        />
      </preventDoubleTap(TouchableOpacity)>
      <AnimatedView>
        <NotificationIcon
          type="token_rewards_summary"
        />
      </AnimatedView>
    </AnimatedView>
    <AnimatedView
      style={
        {
          "alignSelf": "center",
          "flex": 9,
        }
      }
    >
      <MText
        style={
          [
            {
              "fontSize": 15,
              "lineHeight": 18,
            },
            {
              "fontWeight": "400",
            },
            {
              "color": "#b9bbc0",
            },
          ]
        }
      >
        <Merged
          notification={
            NotificationModel {
              "created_timestamp": 1622815259,
              "data": {
                "tokens_formatted": "0.332",
                "usd_formatted": "0.36",
              },
              "entity": {
                "access_id": "2",
                "allow_comments": false,
                "allow_unsubscribed_contact": false,
                "ban_reason": false,
                "banned": "no",
                "boostProPlus": false,
                "boost_autorotate": true,
                "boost_rating": 1,
                "briefdescription": "Full Stack Developer at @minds",
                "btc_address": "1M27rFgyxqGdbmyjABu7J3pqa2YPhFs4mp",
                "canary": true,
                "categories": [],
                "chat": true,
                "city": "Colonia Avellaneda",
                "container_guid": "0",
                "disable_autoplay_videos": true,
                "disabled_boost": false,
                "dismissed_widgets": [
                  "language-sidebar-2020",
                  "discovery-disclaimer-2020",
                ],
                "email_confirmed": true,
                "eth_wallet": "0xd3004e3cbA40079c31F80F0C465aacc34588bd79",
                "fb": false,
                "feature_flags": false,
                "featured_id": false,
                "founder": false,
                "gender": "",
                "guid": "968187695744425997",
                "hashtags": false,
                "hide_share_buttons": true,
                "icontime": "1613843786",
                "is_admin": false,
                "is_mature": false,
                "language": "en",
                "last_accepted_tos": 1558597098,
                "last_avatar_upload": "1565703100",
                "legacy_guid": false,
                "liquidity_spot_opt_out": 0,
                "mature": 1,
                "mature_lock": false,
                "merchant": {
                  "id": "acct_1Gj1tZDf4yhop99j",
                  "service": "stripe",
                },
                "mode": 0,
                "monetized": "",
                "name": "Juan",
                "nsfw": [],
                "nsfw_lock": [],
                "onchain_booster": 0,
                "opted_in_hashtags": 5,
                "owner_guid": "0",
                "p2p_media_enabled": false,
                "pinned_posts": [
                  "1234531362564866048",
                ],
                "plus": false,
                "pro": false,
                "pro_published": false,
                "programs": [],
                "rating": 1,
                "rewards": true,
                "signup_method": false,
                "site_guid": false,
                "social_profiles": [
                  {
                    "key": "linkedin",
                    "value": "linkedin.com/in/juan-manuel-solaro-730a48184",
                  },
                ],
                "subscribed": false,
                "subscriber": false,
                "subtype": false,
                "surge_token": "eyJzZXJ2aWNlIjoiZ29vZ2xlIiwidG9rZW4iOiJkbzZ2LWhfWGhxMDpBUEE5MWJGQ3N0T3c4QzdPR3M0ZjZDZ3VzWWhma0JScGstMF9rN0xValQyUmJhRGtmTmJpZXlSSG1oRmRJeTlRRTlEY0tsV0ZPZFNxR3hfNU9YQnI3VXFUV3paMDFNZ1ZSYm9fVXM3WjZrQVpiX2ZtdDU2akh2LVdUNXk3RzM4ZlBIQ1QwSDZEdXU5RyJ9",
                "tags": [
                  "technology",
                  "crypto",
                  "blockchain",
                  "music",
                  "art",
                ],
                "theme": "dark",
                "time_created": "1556209934",
                "time_updated": false,
                "toaster_notifications": true,
                "type": "user",
                "urn": "urn:user:968187695744425997",
                "username": "juanmsolaro",
                "verified": true,
                "website": "",
                "wire_rewards": {
                  "description": "",
                  "rewards": {
                    "money": [
                      {
                        "amount": "1",
                        "description": "VIP",
                      },
                    ],
                    "points": [],
                    "tokens": [
                      {
                        "amount": "1",
                        "description": "Common Citizen",
                      },
                      {
                        "amount": "2",
                        "description": "Adventurer",
                      },
                      {
                        "amount": "3",
                        "description": "Hero",
                      },
                    ],
                  },
                },
                "yt_channels": [],
              },
              "entity_urn": "urn:user:968187695744425997",
              "from": {
                "_collapsed": false,
                "_viewed": false,
                "access_id": false,
                "allow_comments": false,
                "allow_unsubscribed_contact": false,
                "ban_reason": false,
                "banned": "no",
                "blocked": undefined,
                "boostProPlus": false,
                "boost_autorotate": true,
                "boost_rating": 1,
                "briefdescription": "",
                "btc_address": "",
                "canary": false,
                "categories": [],
                "city": false,
                "confirmEmail": [Function],
                "container_guid": false,
                "disable_autoplay_videos": false,
                "disabled_boost": false,
                "dismissed_widgets": [],
                "email_confirmed": true,
                "eth_wallet": "",
                "fb": false,
                "feature_flags": false,
                "featured_id": false,
                "founder": false,
                "gender": false,
                "groupsCount": 0,
                "guid": "100000000000000519",
                "hashtags": false,
                "hide_share_buttons": false,
                "icontime": "1623077876",
                "impressions": undefined,
                "isOwner": [Function],
                "is_admin": false,
                "is_mature": false,
                "last_accepted_tos": 1558597098,
                "last_avatar_upload": "0",
                "legacy_guid": false,
                "liquidity_spot_opt_out": 0,
                "mature": 0,
                "mature_lock": false,
                "mature_visibility": false,
                "merchant": false,
                "mode": 0,
                "monetized": false,
                "name": "Minds",
                "nsfw": [],
                "nsfw_lock": [],
                "onchain_booster": 0,
                "opted_in_hashtags": 0,
                "owner_guid": false,
                "p2p_media_enabled": false,
                "pending_subscribe": false,
                "permissions": {},
                "pinned_posts": [],
                "plus": false,
                "plus_method": "tokens",
                "pro": false,
                "pro_method": "tokens",
                "pro_published": false,
                "programs": [],
                "rating": 1,
                "rewards": false,
                "signup_method": false,
                "site_guid": false,
                "social_profiles": [],
                "subscribed": true,
                "subscriber": false,
                "subscribers_count": undefined,
                "subscriptions_count": 0,
                "subtype": false,
                "surge_token": "",
                "tags": [],
                "theme": false,
                "time_created": "1623077876",
                "time_updated": false,
                "toaster_notifications": true,
                "type": "user",
                "urn": "urn:user:100000000000000519",
                "username": "minds",
                "verified": false,
                "website": false,
                "wire_rewards": null,
                "yt_channels": [],
              },
              "from_guid": "100000000000000519",
              "merge_key": "52161ea64224a6ac08d653f9dd40b6e568d55362945138174507bf566e9e5e63",
              "merged_count": 0,
              "merged_from": [],
              "merged_from_guids": [],
              "read": true,
              "to_guid": "968187695744425997",
              "type": "token_rewards_summary",
              "urn": "urn:notification:968187695744425997-4a427b80-c53d-11eb-a497-93e2755f275f",
              "uuid": "4a427b80-c53d-11eb-a497-93e2755f275f",
            }
          }
          router={
            {
              "navToEntity": [Function],
            }
          }
        />
         
      </MText>
    </AnimatedView>
    <AnimatedView
      style={
        {
          "alignSelf": "center",
          "flex": 2,
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <MText
        style={
          [
            {
              "fontSize": 15,
              "lineHeight": 18,
            },
            {
              "fontWeight": "400",
            },
            {
              "color": "#b9bbc0",
            },
          ]
        }
      >
        1w
      </MText>
    </AnimatedView>
  </AnimatedView>
  <Memo()
    notification={
      NotificationModel {
        "created_timestamp": 1622815259,
        "data": {
          "tokens_formatted": "0.332",
          "usd_formatted": "0.36",
        },
        "entity": {
          "access_id": "2",
          "allow_comments": false,
          "allow_unsubscribed_contact": false,
          "ban_reason": false,
          "banned": "no",
          "boostProPlus": false,
          "boost_autorotate": true,
          "boost_rating": 1,
          "briefdescription": "Full Stack Developer at @minds",
          "btc_address": "1M27rFgyxqGdbmyjABu7J3pqa2YPhFs4mp",
          "canary": true,
          "categories": [],
          "chat": true,
          "city": "Colonia Avellaneda",
          "container_guid": "0",
          "disable_autoplay_videos": true,
          "disabled_boost": false,
          "dismissed_widgets": [
            "language-sidebar-2020",
            "discovery-disclaimer-2020",
          ],
          "email_confirmed": true,
          "eth_wallet": "0xd3004e3cbA40079c31F80F0C465aacc34588bd79",
          "fb": false,
          "feature_flags": false,
          "featured_id": false,
          "founder": false,
          "gender": "",
          "guid": "968187695744425997",
          "hashtags": false,
          "hide_share_buttons": true,
          "icontime": "1613843786",
          "is_admin": false,
          "is_mature": false,
          "language": "en",
          "last_accepted_tos": 1558597098,
          "last_avatar_upload": "1565703100",
          "legacy_guid": false,
          "liquidity_spot_opt_out": 0,
          "mature": 1,
          "mature_lock": false,
          "merchant": {
            "id": "acct_1Gj1tZDf4yhop99j",
            "service": "stripe",
          },
          "mode": 0,
          "monetized": "",
          "name": "Juan",
          "nsfw": [],
          "nsfw_lock": [],
          "onchain_booster": 0,
          "opted_in_hashtags": 5,
          "owner_guid": "0",
          "p2p_media_enabled": false,
          "pinned_posts": [
            "1234531362564866048",
          ],
          "plus": false,
          "pro": false,
          "pro_published": false,
          "programs": [],
          "rating": 1,
          "rewards": true,
          "signup_method": false,
          "site_guid": false,
          "social_profiles": [
            {
              "key": "linkedin",
              "value": "linkedin.com/in/juan-manuel-solaro-730a48184",
            },
          ],
          "subscribed": false,
          "subscriber": false,
          "subtype": false,
          "surge_token": "eyJzZXJ2aWNlIjoiZ29vZ2xlIiwidG9rZW4iOiJkbzZ2LWhfWGhxMDpBUEE5MWJGQ3N0T3c4QzdPR3M0ZjZDZ3VzWWhma0JScGstMF9rN0xValQyUmJhRGtmTmJpZXlSSG1oRmRJeTlRRTlEY0tsV0ZPZFNxR3hfNU9YQnI3VXFUV3paMDFNZ1ZSYm9fVXM3WjZrQVpiX2ZtdDU2akh2LVdUNXk3RzM4ZlBIQ1QwSDZEdXU5RyJ9",
          "tags": [
            "technology",
            "crypto",
            "blockchain",
            "music",
            "art",
          ],
          "theme": "dark",
          "time_created": "1556209934",
          "time_updated": false,
          "toaster_notifications": true,
          "type": "user",
          "urn": "urn:user:968187695744425997",
          "username": "juanmsolaro",
          "verified": true,
          "website": "",
          "wire_rewards": {
            "description": "",
            "rewards": {
              "money": [
                {
                  "amount": "1",
                  "description": "VIP",
                },
              ],
              "points": [],
              "tokens": [
                {
                  "amount": "1",
                  "description": "Common Citizen",
                },
                {
                  "amount": "2",
                  "description": "Adventurer",
                },
                {
                  "amount": "3",
                  "description": "Hero",
                },
              ],
            },
          },
          "yt_channels": [],
        },
        "entity_urn": "urn:user:968187695744425997",
        "from": {
          "_collapsed": false,
          "_viewed": false,
          "access_id": false,
          "allow_comments": false,
          "allow_unsubscribed_contact": false,
          "ban_reason": false,
          "banned": "no",
          "blocked": undefined,
          "boostProPlus": false,
          "boost_autorotate": true,
          "boost_rating": 1,
          "briefdescription": "",
          "btc_address": "",
          "canary": false,
          "categories": [],
          "city": false,
          "confirmEmail": [Function],
          "container_guid": false,
          "disable_autoplay_videos": false,
          "disabled_boost": false,
          "dismissed_widgets": [],
          "email_confirmed": true,
          "eth_wallet": "",
          "fb": false,
          "feature_flags": false,
          "featured_id": false,
          "founder": false,
          "gender": false,
          "groupsCount": 0,
          "guid": "100000000000000519",
          "hashtags": false,
          "hide_share_buttons": false,
          "icontime": "1623077876",
          "impressions": undefined,
          "isOwner": [Function],
          "is_admin": false,
          "is_mature": false,
          "last_accepted_tos": 1558597098,
          "last_avatar_upload": "0",
          "legacy_guid": false,
          "liquidity_spot_opt_out": 0,
          "mature": 0,
          "mature_lock": false,
          "mature_visibility": false,
          "merchant": false,
          "mode": 0,
          "monetized": false,
          "name": "Minds",
          "nsfw": [],
          "nsfw_lock": [],
          "onchain_booster": 0,
          "opted_in_hashtags": 0,
          "owner_guid": false,
          "p2p_media_enabled": false,
          "pending_subscribe": false,
          "permissions": {},
          "pinned_posts": [],
          "plus": false,
          "plus_method": "tokens",
          "pro": false,
          "pro_method": "tokens",
          "pro_published": false,
          "programs": [],
          "rating": 1,
          "rewards": false,
          "signup_method": false,
          "site_guid": false,
          "social_profiles": [],
          "subscribed": true,
          "subscriber": false,
          "subscribers_count": undefined,
          "subscriptions_count": 0,
          "subtype": false,
          "surge_token": "",
          "tags": [],
          "theme": false,
          "time_created": "1623077876",
          "time_updated": false,
          "toaster_notifications": true,
          "type": "user",
          "urn": "urn:user:100000000000000519",
          "username": "minds",
          "verified": false,
          "website": false,
          "wire_rewards": null,
          "yt_channels": [],
        },
        "from_guid": "100000000000000519",
        "merge_key": "52161ea64224a6ac08d653f9dd40b6e568d55362945138174507bf566e9e5e63",
        "merged_count": 0,
        "merged_from": [],
        "merged_from_guids": [],
        "read": true,
        "to_guid": "968187695744425997",
        "type": "token_rewards_summary",
        "urn": "urn:notification:968187695744425997-4a427b80-c53d-11eb-a497-93e2755f275f",
        "uuid": "4a427b80-c53d-11eb-a497-93e2755f275f",
      }
    }
  />
</TouchableOpacity>
`;
